projects/transformers/experiments/squad.py

```diff
@@ -0,0 +1,126 @@
+#  Numenta Platform for Intelligent Computing (NuPIC)
+#  Copyright (C) 2021, Numenta, Inc.  Unless you have an agreement
+#  with Numenta, Inc., for a separate license for this software code, the
+#  following terms and conditions apply:
+#
+#  This program is free software you can redistribute it and/or modify
+#  it under the terms of the GNU Affero Public License version 3 as
+#  published by the Free Software Foundation.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU Affero Public License for more details.
+#
+#  You should have received a copy of the GNU Affero Public License
+#  along with this program.  If not, see htt"://www.gnu.org/licenses.
+#
+#  http://numenta.org/licenses/
+#
+
+"""
+Finetuning on Squad, not Glue.
+"""
+
+from copy import deepcopy
+
+from transformers import Trainer
+
+from callbacks import TrackEvalMetrics
+from trainer_mixins import QuestionAnsweringMixin
+
+from .finetuning import finetuning_bert_100k_glue_get_info
+
+
+class QuestionAnsweringTrainer(QuestionAnsweringMixin, Trainer):
+    pass
+
+
+debug_bert_squad_base = deepcopy(finetuning_bert_100k_glue_get_info)
+debug_bert_squad_base.update(
+    # Model Args
+    model_name_or_path="bert-base-cased",
+    finetuning=True,
+    task_names=None,
+    task_name="squad",
```

Comment:
There are a lot of hyperparameters are related. If task_name is squad, then dataset_name has to be squad and trainer_class has to be QuestionAnsweringTrainer. It is easy to incur into bugs if we forget to set some of these

We can add some code in run.py to check, and either throw an Exception, or make the changes and throw a Warning. If they were part of the same dataclass parser, another possible solution is to do that check in post_init in the dataclass. 

None of this is required, it is not a big deal, but it would help in avoiding bugs.